daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/BinaryNumberTraits.scala

```diff
@@ -103,6 +103,18 @@ trait PrefixedLengthParserMixin {
         "Prefixed length result must be non-negative after dfdl:prefixIncludesPrefixLength adjustment , but was: %d",
         adjustedLen
       )
+      // do checks on facets expressed on prefixLengthType
+      val optSTRD = plElement.erd.optSimpleTypeRuntimeData
+      if (optSTRD.isDefined) {
+        val strd = optSTRD.get
+        val check = strd.executeCheck(plElement)
+        if (check.isError) {
+          val pe = state.toProcessingError(
+            s"The calculated value of ${prefixedLengthERD.namedQName} ($parsedLen) failed check due to ${check.errMsg}"
```

Comment:
This check is also supposed to happen on unparsing. 

We don't normally validate facets while unparsing (yet), so it might not be quite so easy as the parse side. 
If deferring the unparse feature, please add a JIRA ticket.

Reply:
Added in validation check for CalculatedPrefixedLengthUnparserMixin and KnownPrefixedLengthUnparserMixin